pilot/pkg/config/kube/gateway/controller.go

```diff
@@ -255,21 +276,39 @@ func NewController(
 		inputs.Namespaces,
 		ReferenceGrants,
 		inputs.Secrets,
-		options.DomainSuffix,
+		c.domainSuffix,
 		c.gatewayContext,
 		c.tagWatcher,
 		opts,
 	)
 
+	InferencePoolStatus, InferencePools := InferencePoolCollection(
+		inputs.InferencePools,
+		inputs.Services,
+		inputs.HTTPRoutes,
+		inputs.Gateways,
+		httpRoutesByInferencePool,
+		c,
+		opts,
+	)
+
+	// Create a queue for handling service updates.
+	c.shadowServiceReconciler = controllers.NewQueue("inference pool shadow service reconciler",
+		controllers.WithReconciler(c.reconcileShadowService(InferencePools, inputs.Services)),
+		controllers.WithMaxAttempts(5))
+
+	status.RegisterStatus(c.status, InferencePoolStatus, GetStatus)
```

Comment:
nit (can probably be done in a follow up): maybe we shouldn't do this unless the feature flag is on

Reply:
why only the status?